### PR TITLE
agent: disconnect rpc get_oom_event when destroy_sandbox.

### PR DESCRIPTION
**Pull request by Tim-Zhang**
Otherwise it would block the shutdown of ttrpc.

Fixes: #1524

Signed-off-by: Tim Zhang <tim@hyper.sh>

### CODE REVIEW

**Review by jodh-intel — APPROVED** [review]
Thanks @Tim-Zhang !! :smile:

lgtm

File: src/agent/src/rpc.rs [hunk 1]
```diff
@@ -1040,6 +1040,15 @@ impl protocols::agent_ttrpc::AgentService for AgentService {
         // destroy all containers, clean up, notify agent to exit
         // etc.
         sandbox.destroy().await.unwrap();
+        // Close get_oom_event connection,
+        // otherwise it will block the shutdown of ttrpc.
+        let _ = sandbox
```
Comment (lifupan) [inline src/agent/src/rpc.rs:1045]: why not just drop this channel to notify peer exit?

Comment (Tim-Zhang) [reply src/agent/src/rpc.rs:1045]: because the tx is also hold by others

Comment (lifupan) [reply src/agent/src/rpc.rs:1045]: When you do the destroy sandbox, all of the containers had exited and the run_oom_event_monitor had exited, thus there's no one take the tx except the sandbox itself, thus you can just drop it thus all of the tx copy would dropped.

Comment (Tim-Zhang) [reply src/agent/src/rpc.rs]: done

**Review by liubin — APPROVED** [review]

**Pull request merged**